docs/developer-docs/latest/development/custom-fields.md

```diff
@@ -0,0 +1,339 @@
+      defaultMessage: "Color",
+    },
+    intlDescription: {
+      id: "color-picker.color.description",
+      defaultMessage: "Select any color",
+    } 
+    icon: ColorIcon,
+    components: {
+      Input: async () => import(/* webpackChunkName: "input-component" */ "./Input"),
+    } 
+    options: {
+      base: [
+        /*
+          Declare settings to be added to the "Base settings" section
+          of the field in the Content-Type Builder
+        */ 
+        {
+          sectionTitle: { // Add a "Format" settings section
+            id: 'color-picker.color.section.format',
+            defaultMessage: 'Format',
+          },
+          items: [ // Add settings items to the section
+            {
+              /*
+                Add a "Color format" dropdown
+                to choose between 2 different format options
+                for the color value: hexadecimal or RGBA
+              */
+              intlLabel: {
+                id: 'color-picker.color.format.label',
+                defaultMessage: 'Color format',
+              },
+              name: 'options.format',
+              type: 'select',
+              value: 'hex', // option selected by default
+              options: [ // List all available "Color format" options
+                {
+                  key: 'hex',
+                  value: 'hex',
+                  metadatas: {
+                    intlLabel: {
+                      id: 'color-picker.color.format.hex',
+                      defaultMessage: 'Hexadecimal',
+                    },
+                  },
+                },
+                {
+                  key: 'rgba',
+                  value: 'rgba',
+                  metadatas: {
+                    intlLabel: {
+                      id: 'color-picker.color.format.rgba',
+                      defaultMessage: 'RGBA',
+                    },
+                  },
+                },
+              ],
+            },
+          ],
+        },
+      ],
+      advanced: [
+        /*
+          Declare settings to be added to the "Advanced settings" section
+          of the field in the Content-Type Builder
+        */ 
+      ],
+      validator: args => ({
+        format: yup.string().required({
+          id: 'options.color-picker.format.error',
+          defaultMessage: 'The color format is required',
+        }),
+      })
+      }),
+    },
+  });
+}
+```
+
+:::
+
+::: note
+Relations, media, components, or dynamic zones can't be used as a custom field's `type` parameter.
+:::
+
+### Components
+
+`app.customFields.register()` must pass a `components` object with an `Input` React component to use in the Content Manager's edit view.
+
+::: details Example: Registering an Input component
+
+```js
+// path: ./src/plugins/my-custom-field-plugin/strapi-admin.js
+
+register(app) {
+  app.customFields.register({
+    // …
+    components: {
+      Input: async () => import(/* webpackChunkName: "input-component" */ "./Input"),
```

Comment:
Just curious if you know why `Input` is capitalized. It seems inconsistent with the normal formatting in Strapi code.

Reply:
In React, it's actually common to capitalize the component name :-) You can have a look at the [components used in the Content Manager](https://github.com/strapi/strapi/tree/main/packages/core/admin/admin/src/content-manager/components), for instance.

Reply:
Ok, thanks! I assumed there was a good reason.